mediator/src/main/kotlin/no/nav/dagpenger/rapportering/personregister/mediator/service/PersonService.kt

```diff
@@ -0,0 +1,146 @@
+                    pdlIdentliste.konsoliderPersonerTilGjeldendePerson(gjeldendePerson, personer)
+                    konsoliderArbeidssøkerperioderForGjeldendePerson(gjeldendePerson, personer)
+                }
+                gjeldendePerson.setStatus(gjeldendePerson.vurderNyStatus())
+                if (personRepository.finnesPerson(gjeldendePerson.ident)) {
+                    personRepository.oppdaterPerson(gjeldendePerson)
+                } else {
+                    personRepository.lagrePerson(gjeldendePerson)
+                }
+                return gjeldendePerson
+            } else {
+                logger.warn("Fant ingen gjeldende ident for ${personer.map { it.ident }}")
+                return null
+            }
+        }
+    }
+
+    private fun List<Ident>.konsoliderPersonerTilGjeldendePerson(
+        gjeldendePerson: Person,
+        personer: List<Person>,
+    ) {
+        this
+            .filter { it.ident != gjeldendePerson.ident }
+            .forEach { pdlIdent ->
+                val historiskPerson = personer.firstOrNull { it.ident == pdlIdent.ident }
+                gjeldendePerson.hendelser.addAll(historiskPerson?.hendelser ?: emptyList())
+                historiskPerson?.statusHistorikk?.getAll()?.forEach { it ->
+                    gjeldendePerson.statusHistorikk.put(it.first, it.second)
+                }
+                val arbeidssøkerperioder =
+                    (gjeldendePerson.arbeidssøkerperioder + (historiskPerson?.arbeidssøkerperioder ?: emptyList()))
+                        .map { it.copy(ident = gjeldendePerson.ident) }
+                        .toMutableList()
+
+                gjeldendePerson.arbeidssøkerperioder.clear()
+                gjeldendePerson.arbeidssøkerperioder.addAll(arbeidssøkerperioder.distinctBy { it.periodeId })
+
+                personRepository.slettPerson(pdlIdent.ident)
+
+                gjeldendePerson.apply {
+                    if (!meldeplikt && historiskPerson?.meldeplikt == true) {
+                        meldeplikt = true
+                    }
+                    if (meldegruppe != "DAGP" && historiskPerson?.meldegruppe == "DAGP") {
+                        meldegruppe = historiskPerson.meldegruppe
+                    }
+                }
+            }
+    }
+
+    private fun konsoliderArbeidssøkerperioderForGjeldendePerson(
+        gjeldendePerson: Person,
+        personListe: List<Person>,
+    ) {
+        val overtatteArbeidssøkerperioder =
+            gjeldendePerson.arbeidssøkerperioder.filter {
+                it.avsluttet == null && it.overtattBekreftelse == true
+            }
+        if (overtatteArbeidssøkerperioder.size > 1) {
+            val siste = overtatteArbeidssøkerperioder.maxByOrNull { it.startet }
+            val arbeidssøkerperioder =
+                gjeldendePerson.arbeidssøkerperioder.map { arbeidssøkerperiode ->
+                    if (arbeidssøkerperiode != siste) {
```

Comment:
Hvorfor gjør vi ikke dette med den siste perioden?

Reply:
Siste er den siste/nyeste perioden som vi har markert som overtatt. Vi vil beholde denne fordi det da blir den gjeldende perioden for den gjeldende personen.

Reply:
Jeg oppdaterer navnet til `nyesteOvertattePeriode`